### ISSUE TITLE
Does peft supports the custom setting of trainable parameters(for example, some params in word_embeddings)

### ISSUE THREAD

**Issue by dongdongzhaoUP** [huggingface/peft#2067]
### Feature request

Does peft supports the custom setting of trainable parameters(for example, some params in word_embeddings)

### Motivation

To use the method [EP](https://arxiv.org/abs/2103.15543)

### Your contribution

maybe

**Comment by BenjaminBossan** [comment]
If I understand your request correctly, you would like the option to train only, say, a single entry of the embedding matrix. This is not possible at the moment, neither with PEFT, nor for full fine-tuning. I'm sure there are some tricks out there, maybe masking the gradient or so, but it's not implemented in PEFT.

Having such a possibility would be nice for other purposes than the paper you linked. E.g. it's common to add new tokens to the embedding matrix when fine-tuning a model, but users typically don't want/need to train the whole embedding matrix, just the new tokens. As such, if you have an idea or know of some code that shows how to achieve this, it would be appreciated.

**Comment by dongdongzhaoUP** [comment]
Thank you for your reply. I have also considered the applicable scenarios you mentioned. It is indeed a necessary function.
